### PR TITLE
update to pytket qir 0.13.0

### PR DESCRIPTION
**Pull request by cqc-melf**
# Description

update to pytket-qir 0.13.0, adding the profile option to the generation.

# Related issues

Please mention any github issues addressed by this PR.

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: pytket/extensions/quantinuum/backends/quantinuum.py [hunk 1]
```diff
@@ -1490,10 +1513,29 @@ def _convert_result(
...
-        ]
+
+        try:
+            stacked_array = [
+                [int(resultdict[name][i][-1 - ind]) for name, ind in results_selection]
+                for i in range(n_shots)
+            ]
+        except IndexError:
+            # this is only a temporary solution and not fully working
+            # see issue
```
Comment (yao-cqc) [inline pytket/extensions/quantinuum/backends/quantinuum.py:1524]: Can you explain the issue or add a link to the issue?

File: pytket/extensions/quantinuum/backends/quantinuum.py [hunk 2]
```diff
@@ -858,6 +859,8 @@ def submit_program(
         if wasm_file_handler is not None:
             if self.backend_info and not self.backend_info.misc.get("wasm", False):
                 raise WasmUnsupported("Backend does not support wasm calls.")
+            # body["bytecode_base64"] = wasm_file_handler._wasm_file_encoded
+            # see https://github.com/CQCL/pytket-quantinuum/issues/496
```
[the remark this answers is not in the document]
Comment (cqc-melf) [reply pytket/extensions/quantinuum/backends/quantinuum.py:863]: Yes, using `body["cfl"]` gives a warning, but the suggested `body["bytecode_base64"]` is not yet working, or we are missing details on how to use it. There is an issue linked, I will try to solve this.

File: tests/integration/backend_test.py [hunk 3]
```diff
@@ -20,7 +20,11 @@
 from base64 import b64encode
 from collections import Counter
 from pathlib import Path
-from typing import Any, Callable, cast  # pylint: disable=unused-import
+from typing import (  # pylint: disable=unused-import  # pylint: disable=unused-import
```
[the remark this answers is not in the document]
Comment (cqc-melf) [reply tests/integration/backend_test.py]: cleaned up in [602c72e](https://github.com/CQCL/pytket-quantinuum/pull/499/commits/602c72eb81cc9dd0a7f8ed083fc51ae626a9ddb8)

File: pytket/extensions/quantinuum/backends/quantinuum.py [hunk 4]
```diff
@@ -1032,6 +1035,24 @@ def process_circuits(
                     ).items():
                         for i in range(count):
                             results_selection.append((name, i))
+                elif language == Language.PQIR:
```
[the remark this answers is not in the document]
Comment (cqc-melf) [reply pytket/extensions/quantinuum/backends/quantinuum.py]: Done in [43726c0](https://github.com/CQCL/pytket-quantinuum/pull/499/commits/43726c065f50420e04d745eaedfaa34a96e17439)

File: pytket/extensions/quantinuum/backends/quantinuum.py [hunk 5]
```diff
@@ -835,7 +836,7 @@ def submit_program(
             "name": name or f"{self._label}",
             "count": n_shots,
             "machine": self._device_name,
-            "language": language.value,
+            "language": ["OPENQASM 2.0", "QIR 1.0", "QIR 1.0"][language.value],
```
[the remark this answers is not in the document]
Comment (cqc-melf) [reply pytket/extensions/quantinuum/backends/quantinuum.py]: Done in [43726c0](https://github.com/CQCL/pytket-quantinuum/pull/499/commits/43726c065f50420e04d745eaedfaa34a96e17439)

I did not want to made this function public for the tests, if you think this better should be public and used in the tests, too, happy to update that.

File: pytket/extensions/quantinuum/backends/quantinuum.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cqc-alec) [reply pytket/extensions/quantinuum/backends/quantinuum.py:871]: Hmm, that's not how I would interpret the warning, looks to me like it's coming from Python, not L4?

**Review by cqc-alec — APPROVED** [review]

File: pytket/extensions/quantinuum/backends/quantinuum.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (cqc-alec) [reply pytket/extensions/quantinuum/backends/quantinuum.py]: Agree not to make it public.

**Pull request merged**